src/insertItem.ts

```diff
@@ -0,0 +1,449 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as path from "path";
+import * as vscode from "vscode";
+// tslint:disable-next-line:no-duplicate-imports
+import { commands } from "vscode";
+import { IAzureUserInput } from "vscode-azureextensionui";
+import { Json, templateKeys } from "../extension.bundle";
+import { assetsPath } from "./constants";
+import { DeploymentTemplate } from "./DeploymentTemplate";
+import { ext } from './extensionVariables';
+import { TemplateSectionType } from "./sortTemplate";
+
+const insertCursorText = '[]';
+
+export class QuickPickItem<T> implements vscode.QuickPickItem {
+    public label: string;
+    public value: T;
+    public description: string;
+
+    constructor(label: string, value: T, description: string) {
+        this.label = label;
+        this.value = value;
+        this.description = description;
+    }
+}
+
+export function getItemType(): QuickPickItem<string>[] {
+    let items: QuickPickItem<string>[] = [];
+    items.push(new QuickPickItem<string>("String", "string", "A string"));
+    items.push(new QuickPickItem<string>("Secure string", "securestring", "A secure string"));
+    items.push(new QuickPickItem<string>("Int", "int", "An integer"));
+    items.push(new QuickPickItem<string>("Bool", "bool", "A boolean"));
+    items.push(new QuickPickItem<string>("Object", "object", "An object"));
+    items.push(new QuickPickItem<string>("Secure object", "secureobject", "A secure object"));
+    items.push(new QuickPickItem<string>("Array", "array", "An array"));
+    return items;
+}
+
+function getResourceSnippets(): vscode.QuickPickItem[] {
+    let items: vscode.QuickPickItem[] = [];
+    let snippetPath = path.join(assetsPath, "armsnippets.jsonc");
+    let content = fse.readFileSync(snippetPath, "utf8");
+    let tree = Json.parse(content);
```

Comment:
nit: Would have used plain old regular JavaScript JSON parsing (since you don't need to analyze errors/etc), but this is fine too.